clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1214,7 +1214,8 @@ void KernelObjVisitor::visitRecord(const CXXRecordDecl *Owner, ParentTy &Parent,
                                    const CXXRecordDecl *Wrapper,
                                    QualType RecordTy,
                                    HandlerTys &... Handlers) {
-  if (RecordTy->getAsRecordDecl()->hasAttr<SYCLRequiresDecompositionAttr>()) {
+  RecordDecl *RD = RecordTy->getAsRecordDecl();
+  if (RD && RD->hasAttr<SYCLRequiresDecompositionAttr>()) {
```

Comment:
This should be an assert here, not a test.  If we are here and RD is null, something very wrong has happened.